workflow/automation/lib/MgmtDB.py

```diff
@@ -319,23 +331,23 @@ def add_retries(self, n_max_retries: int):
         n_max_retries: The maximum number of retries a task can have"""
         with connect_db_ctx(self._db_file) as cur:
             errored = cur.execute(
-                "SELECT run_name, proc_type "
+                "SELECT run_name, proc_type, state "
                 "FROM state, status_enum "
                 "WHERE state.status = status_enum.id "
                 "AND (status_enum.state  = 'failed' "
                 "OR status_enum.state = 'killed_WCT')"
             ).fetchall()
 
         failure_count = {}
-        for run_name, proc_type in errored:
+        for run_name, proc_type, state in errored:
             key = "{}__{}".format(run_name, proc_type)
             if key not in failure_count.keys():
-                failure_count.update({key: 0})
-            failure_count[key] += 1
+                failure_count.update({key: {"killed_WCT": 0, "failed": 0}})
```

Comment:
In future if we add more failure modes we can abstract the state dictionary generation out. This works for now though